### PR TITLE
Line analysis tests

### LINKED ISSUE #1279: [BUG] Specviz: Line analysis does not work on two subsets using API
**Describe the bug**
A test in #1275 exposed a bug where if the continuum subset is set to subset 1 and spectral subset is set to subset 2, a traceback is returned.

<details>

```
        plugin.spectral_subset_selected = 'Subset 2'
>       plugin.continuum_subset_selected = 'Subset 1'

test_line_analysis.py:263: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:606: in __set__
    self.set(obj, value)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:595: in set
    obj._notify_trait(self.name, old_value, new_value)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:1219: in _notify_trait
    self.notify_change(Bunch(
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/ipywidgets/widgets/widget.py:686: in notify_change
    super(Widget, self).notify_change(change)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:1229: in notify_change
    return self._notify_observers(change)
/Users/javerbukh/opt/anaconda3/envs/jdaviz-dev-2.5/lib/python3.8/site-packages/traitlets/traitlets.py:1266: in _notify_observers
    c(event)
../line_analysis.py:275: in _calculate_statistics
    right_max = np.nanmax([np.nanmax(mark_x['right']), sr.upper.value])
<__array_function__ internals>:180: in nanmax
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = array([nan, nan]), axis = None, out = None, keepdims = <no value>, initial = <no value>, where = <no value>

    @array_function_dispatch(_nanmax_dispatcher)
    def nanmax(a, axis=None, out=None, keepdims=np._NoValue, initial=np._NoValue,
               where=np._NoValue):
        """
        Return the maximum of an array or maximum along an axis, ignoring any
        NaNs.  When all-NaN slices are encountered a ``RuntimeWarning`` is
        raised and NaN is returned for that slice.
    
        Parameters
        ----------
        a : array_like
            Array containing numbers whose maximum is desired. If `a` is not an
            array, a conversion is attempted.
        axis : {int, tuple of int, None}, optional
            Axis or axes along which the maximum is computed. The default is to compute
            the maximum of the flattened array.
        out : ndarray, optional
            Alternate output array in which to place the result.  The default
            is ``None``; if provided, it must have the same shape as the
            expected output, but the type will be cast if necessary. See
            :ref:`ufuncs-output-type` for more details.
    
            .. versionadded:: 1.8.0
        keepdims : bool, optional
            If this is set to True, the axes which are reduced are left
            in the result as dimensions with size one. With this option,
            the result will broadcast correctly against the original `a`.
    
            If the value is anything but the default, then
            `keepdims` will be passed through to the `max` method
            of sub-classes of `ndarray`.  If the sub-classes methods
            does not implement `keepdims` any exceptions will be raised.
    
            .. versionadded:: 1.8.0
        initial : scalar, optional
            The minimum value of an output element. Must be present to allow
            computation on empty slice. See `~numpy.ufunc.reduce` for details.
    
            .. versionadded:: 1.22.0
        where : array_like of bool, optional
            Elements to compare for the maximum. See `~numpy.ufunc.reduce`
            for details.
    
            .. versionadded:: 1.22.0
    
        Returns
        -------
        nanmax : ndarray
            An array with the same shape as `a`, with the specified axis removed.
            If `a` is a 0-d array, or if axis is None, an ndarray scalar is
            returned.  The same dtype as `a` is returned.
    
        See Also
        --------
        nanmin :
            The minimum value of an array along a given axis, ignoring any NaNs.
        amax :
            The maximum value of an array along a given axis, propagating any NaNs.
        fmax :
            Element-wise maximum of two arrays, ignoring any NaNs.
        maximum :
            Element-wise maximum of two arrays, propagating any NaNs.
        isnan :
            Shows which elements are Not a Number (NaN).
        isfinite:
            Shows which elements are neither NaN nor infinity.
    
        amin, fmin, minimum
    
        Notes
        -----
        NumPy uses the IEEE Standard for Binary Floating-Point for Arithmetic
        (IEEE 754). This means that Not a Number is not equivalent to infinity.
        Positive infinity is treated as a very large number and negative
        infinity is treated as a very small (i.e. negative) number.
    
        If the input has a integer type the function is equivalent to np.max.
    
        Examples
        --------
        >>> a = np.array([[1, 2], [3, np.nan]])
        >>> np.nanmax(a)
        3.0
        >>> np.nanmax(a, axis=0)
        array([3.,  2.])
        >>> np.nanmax(a, axis=1)
        array([2.,  3.])
    
        When positive infinity and negative infinity are present:
    
        >>> np.nanmax([1, 2, np.nan, np.NINF])
        2.0
        >>> np.nanmax([1, 2, np.nan, np.inf])
        inf
    
        """
        kwargs = {}
        if keepdims is not np._NoValue:
            kwargs['keepdims'] = keepdims
        if initial is not np._NoValue:
            kwargs['initial'] = initial
        if where is not np._NoValue:
            kwargs['where'] = where
    
        if type(a) is np.ndarray and a.dtype != np.object_:
            # Fast, but not safe for subclasses of ndarray, or object arrays,
            # which do not implement isnan (gh-9009), or fmax correctly (gh-8975)
            res = np.fmax.reduce(a, axis=axis, out=out, **kwargs)
            if np.isnan(res).any():
>               warnings.warn("All-NaN slice encountered", RuntimeWarning,
                              stacklevel=3)
E               RuntimeWarning: All-NaN slice encountered


```

</details>

**To Reproduce**
1. Go to `test_line_analysis`
2. Uncomment the line after the `#TODO` statement
3. Run `pytest -s .` from within the line analysis tests directory

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to add unit test coverage of the line analysis plugin.

Close #1279

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by javerbukh** [comment]
Manually checked coverage, it went from 79.9% to 91%. I'll see if I can cover some more lines before this is ready for merge.

**Review by kecnry — APPROVED** [review]
The remaining (8) uncovered lines are not critical and likely require different datasets or workflows not easily implemented here (with the exception of the one block that likely contains a logic bug that should be fixed in a follow-up issue).

I know we haven't been consistent with this, but if the assertion comparisons to results are just against current values and haven't been validated, maybe we should add a comment stating that to avoid any perception that those are known true values?

File: jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py [hunk 1]
```diff
@@ -94,3 +115,242 @@ def test_coerce_unit():
...
+
+    plugin = specviz_helper.app.get_tray_item_from_name('specviz-line-analysis')
+    assert 'Subset 1' in plugin.spectral_subset.labels
+    assert 'Subset 2' in plugin.spectral_subset.labels
+
+    # TODO: Find why uncommenting the following line
+    #  causes a ValueError: zero-size array
+    # plugin.spectral_subset_selected = 'Subset 2'
+    plugin.continuum_subset_selected = 'Subset 1'
+    plugin.width = 3
```
[the remark this answers is not in the document]
Comment (javerbukh) [reply jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py:264]: Follow-up ticket is #1279. In response to your second comment, can value validation also be outside of the scope of this PR? That seems like something the POs will have to be involved with.

[the remark this answers is not in the document]
Comment (kecnry) [reply jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py:264]: Yes, for sure, but I think adding in-line comments in the test stating the values are not validated would help ensure we don't accidentally think the numbers came from somewhere other than the current results.

[the remark this answers is not in the document]
Comment (javerbukh) [reply jdaviz/configs/specviz/plugins/line_analysis/tests/test_line_analysis.py:266]: Done!

**Review by pllim — APPROVED** [review]
I applied my minor review comments straight to the PR, so LGTM now. 😸

**Comment by javerbukh** [comment]
Tests pass for me locally, thank you both!

**Pull request merged**